### PR TITLE
adds started extension concepts article - to be fleshed out

### PR DESCRIPTION
**Pull request by haydnba**

### CODE REVIEW

File: articles/browser-extension-basics.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+## Browser Extension Basics
+
+Useful Resources:
+
+- [MDN Anatomy of an Extension](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension)
```
Comment (porkbrain) [inline articles/browser-extension-basics.md]: I'd suggest having the links at the bottom of the document.

```
<!-- Invisible List of References -->
[mozilla-anatomy-webext]: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Anatomy_of_a_WebExtension
```

Then you can use it with `[MDN Anatomy of an Extension][mozilla-anatomy-webext]`.

Comment (haydnba) [reply articles/browser-extension-basics.md]: Ah cool thanks! Pav said he'd fill me in on some of the Jekyll syntax

File: articles/browser-extension-basics.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - specify extension metadata, e.g. `name`, `version`, `locale` etc.
+    - point to core files, e.g. `background`, `content_scripts`, `devtools_page`, `options` and bundled `web_accessible_resources`
+    - assert permissions on browser APIs and other resources, urls etc. e.g.`storage`, `tabs`, `bookmarks`, `https://*/*`
+    - declare UI elements e.g. `browser_action` icon
+
+2. Content ([Firefox](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts), [Chrome](https://developer.chrome.com/extensions/content_scripts)): The content scripts (`.js` or `.css`) are loaded into and executed within the context of a given web page. They have access to the DOM and as such can read and manipulate the existing HTML. These scripts can dispatch XHRs. In order to interact with most other browser APIs/resources or with other extension elements (e.g. to tell them about certain DOM elements present) they must usually communicate via the intermediary of a `background script`, through message passing (see below). In order to communicate with a page `<script>` (or vice-versa) the `window.postMessage()` API must be used.
```
Comment (porkbrain) [inline articles/browser-extension-basics.md]: > through message passing

I'd also include a link to the message passing article.

File: articles/browser-extension-basics.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+3. Background: The background script(s) (`.js`, or a single `.html` background page) are loaded into and execute within the browser environment. They access most of the browser APIs (depending on their permissions as specified in `manifest.json`) and can listen to browser events etc. Background scripts tend either to run for an extended duration in order to maintain state or to serve as a semi-dormant event handler. They are provided their own `window` for DOM access (with limitations) and can dispatch XHRs. The background scripts cannot interact directly with a loaded webpage, but must act via a content script that they load into that page context (see above).
```
Comment (porkbrain) [inline articles/browser-extension-basics.md]: typo "executed"

**Review by porkbrain — APPROVED** [review]

File: articles/browser-extension-basics.md (hunk 2 above)
Comment (haydnba) [reply articles/browser-extension-basics.md]: 👍 will do

File: articles/browser-extension-basics.md (hunk 3 above)
Comment (haydnba) [reply articles/browser-extension-basics.md]: 👍 thx

File: articles/browser-extension-basics.md (hunk 2 above)
Comment (kouks) [inline articles/browser-extension-basics.md]: Also maybe mention that you need to add permissions to the manifest to be able to inject a content script.

Comment (haydnba) [reply articles/browser-extension-basics.md]: 👍

**Review by kouks — APPROVED** [review]

**Pull request merged**